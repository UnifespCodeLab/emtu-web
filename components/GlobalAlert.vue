<template>
  <v-alert
    class="global-alert"
    dismissible
    transition="scale-transition"
    :type="alertType"
    :value="isVisible"
  >
    {{ alertMessage }}
  </v-alert>
</template>

<script>
import { mapState, mapActions } from 'vuex'

export default {
  name: 'GlobalAlert',
  computed: {
    ...mapState('alert', ['isVisible', 'alertMessage', 'alertType'])
  },

  watch: {
    isVisible (currentValue) {
      if (currentValue === true) {
        setTimeout(() => this.hideAlert(), 5000)
      }
    }
  },
  methods: {
    ...mapActions('alert', ['hideAlert'])
  }
}
</script>

<style lang="scss" scoped>
.global-alert {
  position: absolute;
  right: 0;
}
</style>
