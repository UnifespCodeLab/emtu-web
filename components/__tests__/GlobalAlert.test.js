import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import GlobalAlert from '@/components/GlobalAlert.vue'

describe('GlobalAlert', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  let vuetify
  let store
  let actions
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    actions = {
      hideAlert: jest.fn()
    }
    store = new Vuex.Store({
      modules: {
        alert: {
          namespaced: true,
          state: {
            isVisible: true,
            alertMessage: 'Teste',
            alertType: 'warning',
          },
          actions
        }
      }
    })

    wrapper = mount(GlobalAlert, {
      localVue,
      vuetify,
      store
    })
  })

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render correct message', () => {
    const message = wrapper.findComponent({ name: 'VAlert' })
    expect(message.text()).toBe('Teste')
  })

  it('should render message type correctly', () => {
    const alert = wrapper.findComponent({ name: 'VAlert' })
    expect(alert.props().type).toBe('warning')
  })

  it('should call hideAlert method directly', async () => {
    wrapper.vm.hideAlert();
    await wrapper.vm.$nextTick();
    expect(actions.hideAlert).toHaveBeenCalled();
  })
})
