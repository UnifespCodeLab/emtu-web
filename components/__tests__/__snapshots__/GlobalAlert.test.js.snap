// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalAlert matches the snapshot 1`] = `
<transition-stub
  class="global-alert"
  name="scale-transition"
>
  <div
    class="v-alert v-sheet theme--dark warning"
    role="alert"
  >
    <div
      class="v-alert__wrapper"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate v-alert__icon mdi mdi-exclamation theme--dark"
      />
      <div
        class="v-alert__content"
      >
        
  Teste

      </div>
      <button
        aria-label="Close"
        class="v-alert__dismissible v-btn v-btn--icon v-btn--round theme--dark v-size--small"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-close-circle theme--dark"
          />
        </span>
      </button>
    </div>
  </div>
</transition-stub>
`;
